pkg/api/owners.go

```diff
@@ -0,0 +1,735 @@
+}
+
+func (o *OwningPlugins) MemUseHierarchyOwner(id string) (string, bool) {
+	return o.OwnersFor(id).SimpleOwner(FieldMemUseHierarchy)
+}
+
+func (o *OwningPlugins) CpuSharesOwner(id string) (string, bool) {
+	return o.OwnersFor(id).SimpleOwner(FieldCpuShares)
+}
+
+func (o *OwningPlugins) CpuQuotaOwner(id string) (string, bool) {
+	return o.OwnersFor(id).SimpleOwner(FieldCpuQuota)
+}
+
+func (o *OwningPlugins) CpuPeriodOwner(id string) (string, bool) {
+	return o.OwnersFor(id).SimpleOwner(FieldCpuPeriod)
+}
+
+func (o *OwningPlugins) CpuRealtimeRuntimeOwner(id string) (string, bool) {
+	return o.OwnersFor(id).SimpleOwner(FieldCpuRealtimeRuntime)
+}
+
+func (o *OwningPlugins) CpuRealtimePeriodOwner(id string) (string, bool) {
+	return o.OwnersFor(id).SimpleOwner(FieldCpuRealtimePeriod)
+}
+
+func (o *OwningPlugins) CpusetCpusOwner(id string) (string, bool) {
+	return o.OwnersFor(id).SimpleOwner(FieldCpusetCpus)
+}
+
+func (o *OwningPlugins) CpusetMemsOwner(id string) (string, bool) {
+	return o.OwnersFor(id).SimpleOwner(FieldCpusetMems)
+}
+
+func (o *OwningPlugins) PidsLimitOwner(id string) (string, bool) {
+	return o.OwnersFor(id).SimpleOwner(FieldPidsLimit)
+}
+
+func (o *OwningPlugins) HugepageLimitOwner(id, size string) (string, bool) {
+	return o.OwnersFor(id).CompoundOwner(FieldHugepageLimits, size)
+}
+
+func (o *OwningPlugins) BlockioClassOwner(id string) (string, bool) {
+	return o.OwnersFor(id).SimpleOwner(FieldBlockioClass)
+}
+
+func (o *OwningPlugins) RdtClassOwner(id string) (string, bool) {
+	return o.OwnersFor(id).SimpleOwner(FieldRdtClass)
+}
+
+func (o *OwningPlugins) CgroupsUnifiedOwner(id, key string) (string, bool) {
+	return o.OwnersFor(id).CompoundOwner(FieldCgroupsUnified, key)
+}
+
+func (o *OwningPlugins) CgroupsPathOwner(id string) (string, bool) {
+	return o.OwnersFor(id).SimpleOwner(FieldCgroupsPath)
+}
+
+func (o *OwningPlugins) OomScoreAdjOwner(id string) (string, bool) {
+	return o.OwnersFor(id).SimpleOwner(FieldOomScoreAdj)
+}
+
+func (o *OwningPlugins) RlimitOwner(id, typ string) (string, bool) {
+	return o.OwnersFor(id).CompoundOwner(FieldRlimits, typ)
+}
+
+func (o *OwningPlugins) OwnersFor(id string) *FieldOwners {
+	f, ok := o.Owners[id]
+	if !ok {
+		f = NewFieldOwners()
+		o.Owners[id] = f
+	}
+	return f
+}
+
+func NewFieldOwners() *FieldOwners {
+	return &FieldOwners{
+		Simple:   make(map[int32]string),
+		Compound: make(map[int32]*CompoundFieldOwners),
+	}
+}
+
+func (f *FieldOwners) ClaimCompound(field int32, key, plugin string) error {
+	m, ok := f.Compound[field]
+	if !ok {
+		m = NewCompoundFieldOwners()
+		f.Compound[field] = m
+	}
+
+	if other, claimed := m.Owners[key]; claimed {
+		return f.Conflict(field, plugin, other, key)
+	}
+
+	m.Owners[key] = plugin
+	return nil
+}
+
+func (f *FieldOwners) ClaimSimple(field int32, plugin string) error {
+	other, claimed := f.Simple[field]
+	if claimed {
+		return f.Conflict(field, plugin, other)
+	}
+
+	f.Simple[field] = plugin
+	return nil
+}
```

Comment:
Just a note here that this is not thread-safe.  If we are guaranteed to never call this from multiple goroutines that's probably fine, but it's unclear whether that's the case.

Reply:
Those claiming functions are always called from a single goroutine. But I'll add a clarifying comment, and a warning, about it to owners.go.